tests/writeAPI/extensionPropertyAbstract.php

```diff
@@ -0,0 +1,710 @@
+<?php
+/*
+ * Copyright (C) 2020 STFC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http:#www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+require_once __DIR__ . '/abstractClass.php';
+
+/**
+ * Abstract class for tests of methods for entities that have extension properties
+ *
+ */
+abstract class extensionPropertyAbstract extends AbstractWriteAPITestClass{
+  private $sampleExtensionProps =  array('someKey' => 'someValue', 'SOmEOTHer  Key'=>'some valuE', 'Another .-_@Key'=>'some .-_@value' );
+  private $sampleExtensionPropsWDup =  array('someKey' => 'This is a duplicate', 'New Key 1'=>'val', 'New Key 2'=>'val' );
+  private $sampleExtensionPropKey = 'Some. -@_Key';
+  private $sampleExtensionPropVal = 'Some. -@_ Value';
+  private $sampleExtensionPropNewVal = 'Some. -@_ new Value';
+
+  /*
+   *This function provides each instance of the class a method to feed in an instance
+   *of the type of entity the class was created to test the methods for.
+   */
+  abstract protected function getSampleEntity();
+
+  /*
+   *This function provides each instance of the class a method to feed in a strings
+   *saying ehat type of entity it is.
+   */
+  abstract protected function getSampleEntityType();
+
+  /*
+   * Implements abstract function. A well formatted PI request for an entity
+   * type that exists should be rejected if unauthenticated (as we don't
+   * currently support GET requests). Extension properties are as good a test for
+   * this as any.
+   */
+  public function test_APIUnauthenticated(){
+    print __METHOD__ . "\n";
+
+    $entity = $this->getSampleEntity();
+    $entType = $this->getSampleEntityType();
+
+    foreach (array('POST','PUT','DELETE') as $method) {
+      #The write API index uses null for the authentication identifier if the user is unauthenticated
+      $APIOutput = $this->wellFormatedWriteAPICall ($method, json_encode($this->sampleExtensionProps), null, $entType, $entity->getId(), 'extensionproperties');
+
+      #As it is unauthenticated it should return a 403
+      $this->assertequals(403,$APIOutput['httpResponseCode']);
+
+      #Check that creating wasn't successful
```

Comment:
```suggestion
      #Check that creating was unsuccessful
```
Minor, but I think it implies that it *should* fail, rather than it being an accident.